src/components/Button.Types.ts

```diff
@@ -0,0 +1,14 @@
+export const buttonVariants = [
+  'primary',
+  'secondary',
+  'tertiary',
+  'action-menu',
+  'action-menu-secondary',
+  'action-menu-muted',
+  'plain',
+  'plain-dangerous',
+  'full',
+] as const
+export type ButtonVariant = typeof buttonVariants[number]
+export const buttonTypes = ['button', 'submit', 'reset'] as const
+export type ButtonType = typeof buttonTypes[number]
```

Comment:
I think I'd prefer these to be defined in the component SFC itself, otherwise it has the effect you pointed out a while ago (at least I think it was you, might have been @krysal maybe?) that it splits the prop type information from the place where you'd need it most, in the component code itself.

Though I guess with TS you can find it more easily than before, either in the auto-complete type information or with "go to definition" in an editor.

If you decide to leave it in separate files (which is fine with me in the end) can we exclude `*.types.ts` from the title case naming ESLint rule so that we don't have the weird `*.Types.ts` situation :sweat_smile: 

It should just be a matter of updating [this line](https://github.com/WordPress/openverse-frontend/blob/3922733043c1616944934f3997f15ca14f5f28b4/src/components/.eslintrc.js#L9) to include TypeScript files.

Reply:
Yes, it was me complaining about not understanding what props a component gets when the props are in a different file :) For `buttonTypes`, I had to extract it because the types are also used by `VIconButton`. I will check the `buttonVariants`, and move them back to SFC if they are not used anywhere else.

Thank you so much for linking the line in ESLint config, I tried to find it this morning, but couldn't :)

Reply:
Is it possible to import types from a vue SFC? Given that `VIconButton` already imports from `VButton`, if we export the `types` type from the button component itself it'd be convenient (and help localize the types) to get it from the same place as the component.

Reply:
Done! This is much better!